### PR TITLE
fix particle system memory issue

### PR DESCRIPTION
**Pull request by SantyWang**
Re: #

Changelog:
 *

### CODE REVIEW

File: cocos/core/assets/material.ts [hunk 1]
```diff
@@ -483,6 +481,7 @@ export class Material extends Asset {
         this.initialize({
             effectName: 'unlit',
             defines: { USE_COLOR: true },
+            technique: 0,
```
Comment (SantyWang) [inline cocos/core/assets/material.ts:484]: only can use technique 0 of unlit

File: cocos/particle/models/particle-batch-model.ts [hunk 2]
```diff
@@ -322,8 +320,12 @@ export default class ParticleBatchModel extends scene.Model {
 
     public destroy () {
         super.destroy();
+        this._vertAttrs = null;
         this._vBuffer = null;
         this._vdataF32 = null;
+        this._vdataUint32 = null;
+        this._material = null;
```
Comment (SantyWang) [inline cocos/particle/models/particle-batch-model.ts:327]: reset to null

File: cocos/particle/renderer/particle-system-renderer-cpu.ts [hunk 3]
```diff
@@ -179,6 +179,11 @@ export default class ParticleSystemRendererCPU extends ParticleSystemRendererBas
         this.setVertexAttributes();
     }
 
+    public onDestroy (): void {
+        this._particles?.destroy();
```
Comment (SantyWang) [inline cocos/particle/renderer/particle-system-renderer-cpu.ts:183]: destroy particle pool

**Review by zxx43 — APPROVED** [review]

**Review by pandamicro — APPROVED** [review]

**Comment by SantyWang** [comment]
https://github.com/cocos/3d-tasks/issues/12262

**Pull request merged**